docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,160 @@
+# Agent-Based Installer Authentication and Authorization - OpenShift 4.18
+
+## Introduction
+Since OpenShift 4.18, the Assisted Service (AS) APIs used by the Agent-Based Installer (ABI) have been secured through authentication and authorization mechanisms. This added security ensures that only authorized services can interact with AS. This document provides an overview of how the authentication and authorization mechanisms implemented for ABI.
+
+## Enabling Authentication and Authorization
+
+To enable authentication and authorization, the `AuthType` setting must be configured in the `assisted-service.env.template` file. Set the following value:
+
+```bash
+AuthType = "agent-installer-local"
+```
+
+This setting ensures that the authentication mechanism required by the agent-based installer is activated and is an internal configuration.
+
+## Tokens for Agent-Based Installer
+
+When creating an Agent ISO (or PXE artifacts) for installing a new cluster, ABI generates three separate authentication tokens: `agentAuth`, `userAuth`, and `watcherAuth`. These tokens serve different purposes and have varying levels of access. The tokens are the part of the ignition that gets embeded in the CoreOS base ISO along with other ABI specific configurations for a cluster. The tokens are stored `/usr/local/share/assisted-service/assisted-service.env` in the ISO. The base64 encoded public key required to validate these tokens is also saved in the same file. Additionally, this data also gets stored in a hidden file `.openshift_state_install.json` present in the directory location mentioned when creating ISO.
```

Comment:
```suggestion
When creating an Agent ISO (or PXE artifacts) for installing a new cluster, ABI generates three separate authentication tokens: `agentAuth`, `userAuth`, and `watcherAuth`. These tokens serve different purposes and have varying levels of access. The tokens are the part of the ignition that gets embeded in the CoreOS base ISO along with other ABI specific configurations for a cluster. The tokens are stored in `/usr/local/share/assisted-service/assisted-service.env` in the ISO. The base64 encoded public key required to validate these tokens is also saved in the same file. Additionally, this data also gets stored in a hidden file `.openshift_state_install.json` present in the directory location mentioned when creating ISO.
```
Why is it stored in the hidden file?

Reply:
The tokens are part of the ignition and is also an asset hence those are saved in the hidden file .openshift_state_install.json like all the assets and other data.